Plugins/Creature/Extras/BonusStacking.cpp

```diff
@@ -0,0 +1,569 @@
+                    }
+                }
+            }
+
+            nEffectBonus = !s_nSavingThrowStackingMode ? GetStackedBonus(false) : GetUnstackedBonus(false, s_nSavingThrowStackingMode);
+            nEffectPenalty = s_bAlwaysStackPenalties || !s_nSavingThrowStackingMode ? GetStackedBonus(true) : GetUnstackedBonus(true, s_nSavingThrowStackingMode);
+
+            {
+                uint32_t nSavingThrowBonusLimit = Globals::AppManager()->m_pServerExoApp->GetSavingThrowBonusLimit();
+                nEffectBonus = std::min(nEffectBonus, nSavingThrowBonusLimit);
+                nEffectPenalty = std::min(nEffectPenalty, nSavingThrowBonusLimit);
+            }
+
+            if (thisPtr->m_pStats->HasFeat(Constants::Feat::SacredDefense1))
+            {
+                int nChampionLevel = thisPtr->m_pStats->GetNumLevelsOfClass(Constants::ClassType::DivineChampion);
+                if (nChampionLevel > 1)
+                    nEffectBonus += nChampionLevel >> 1; //Level / 2
+            }
+
+            return nEffectBonus - nEffectPenalty;
+
+        case 4:
+            for (int i = thisPtr ? thisPtr->m_pStats->m_nAbilityPtr : 0;
+                i < thisPtr->m_appliedEffects.num
+                && thisPtr->m_appliedEffects.element[i]->m_nType >= Constants::EffectTrueType::AbilityIncrease
+                && thisPtr->m_appliedEffects.element[i]->m_nType <= Constants::EffectTrueType::AbilityDecrease;
+                i++)
+            {
+                auto pEffect = thisPtr->m_appliedEffects.element[i];
+                uint8_t nEffectAbility = pEffect->GetInteger(0);
+
+                if (nEffectAbility != nAbilityScore)
+                    continue;
+
+                uint32_t nEffectStrength = pEffect->GetInteger(1);
+
+                if (pEffect->m_nType == Constants::EffectTrueType::AbilityIncrease)
+                {
+                    if (auto pCreator = Globals::AppManager()->m_pServerExoApp->GetItemByGameObjectID(pEffect->m_oidCreator))
+                        AddPositiveEffectOID(pCreator->m_idSelf, nEffectStrength);
+                    else
+                        AddPositiveEffect(pEffect->m_nSpellId, nEffectStrength);
+                }
+                else if (pEffect->m_nType == Constants::EffectTrueType::AbilityDecrease)
+                {
+                    if (auto pCreator = Globals::AppManager()->m_pServerExoApp->GetItemByGameObjectID(pEffect->m_oidCreator))
+                        AddNegativeEffectOID(pCreator->m_idSelf, nEffectStrength);
+                    else
+                        AddNegativeEffect(pEffect->m_nSpellId, nEffectStrength);
+                }
+            }
+
+            nEffectBonus = !s_nAbilityStackingMode ? GetStackedBonus(false) : GetUnstackedBonus(false, s_nAbilityStackingMode);
+            nEffectPenalty = s_bAlwaysStackPenalties || !s_nAbilityStackingMode ? GetStackedBonus(true) : GetUnstackedBonus(true, s_nAbilityStackingMode);
```

Comment:
Please add parentheses to make priorities obvious. Took me a minute to figure it out :)